### DISCUSSION TITLE
[A/L] (2021/22) Funzioni - Esercizio 1

### DISCUSSION THREAD

**Discussion by matypist** [sapienzastudentsnetwork/metodi-matematici-per-l-informatica#241] [Esercizi A-L]
![image](https://user-images.githubusercontent.com/69389461/152314736-2b5eff27-1546-425e-b429-30171b17acae.png)

**Comment by LeuxLello** [comment]
In teoria è vero perchè considerando la definizione di funzione suriettiva abbiamo che ogni elemento del codominio ha almeno una freccia entrante, quindi ogni y appartenente a Y ha almeno una freccia da X. Una funzione è invece iniettiva perchè  ogni elemento del codominio ha al MASSIMO una freccia entrante dal dominio. quindi come esposto qua sotto, se invertiamo dominio e codominio avremo che la funzione Y--->X è [iniettiva]. nel disegno qui sotto ho semplicemente invertito 
i due insiemi

![Senza titolo](https://user-images.githubusercontent.com/104726379/172909946-183d6f24-24c2-47e0-a7e6-8df140ff567c.png)

**Comment by Rurik-D** [comment]
Vero.
L'esercizio non parla né di invertibilità né di identità, ciò significa che la funzione g(y) non è in alcun modo legata alla funzione f(x) se non per il suo codominio. Negare l'affermazione dell'esercizio significherebbe affermare che data una qualsiasi funzione suriettiva non possa esistere una qualsiasi funzione iniettiva che possa lavorare sugli stessi insiemi solamente invertiti. Allego l'immagine di una dimostrazione grafica.
![esempio es1 funzioni](https://user-images.githubusercontent.com/96311685/185612244-6ebcbac0-fbcf-44cd-9f59-3299e364cb13.png)

**Comment by LuginiAndrea** [comment]
Prendiamo una funzione h: X -> Y:
- Se #X > #Y allora la funzione non può essere iniettiva in quanto esiste almeno una coppia di valori x ai quali dovremmo associare lo stesso y
- Se #Y > #X la funzione non può essere suriettiva in quanto, poichè ad ogni x corrisponde un solo y, vuol dire che esistono degli y per cui non vale y = h(x)

Sapendo che f: X -> Y è suriettiva possiamo concludere che #X >= #Y, ovvero #Y <= #X.
Di conseguenza presa g: Y -> X, poichè rispettiamo il vincolo che la cardinalità dell'insieme di partenza sia minore o uguale a quella dell'insieme di arrivo, g può essere iniettiva
